src/Engine/Manager/ScriptManager.cpp

```diff
@@ -424,12 +429,113 @@ void ScriptManager::BuildAllScripts() {
     }
 }
 
+void ScriptManager::FillPropertyMap(Script* script) {
+
+    BuildScript(script->scriptFile);
+    CreateInstance(script);
+
+    int propertyCount = script->instance->GetPropertyCount();
+
+    for (int n = 0; n < propertyCount; n++) {
+
+        int typeId = script->instance->GetPropertyTypeId(n);
+        void *varPointer = script->instance->GetAddressOfProperty(n);
+
+        std::map<std::string, std::map<int, void*>>::iterator it = script->propertyMap.find(script->instance->GetPropertyName(n));
```

Comment:
If you want you can replace the verbose type with auto to make things more readable :)